pkgs/build-support/fetchzip/default.nix

```diff
@@ -23,22 +23,31 @@ lib.overrideDerivation (fetchurl ({
     ''
       export PATH=${unzip}/bin:$PATH
       mkdir $out
-      cd $out
-      renamed="$TMPDIR/${baseNameOf url}"
+
+      unpackDir="$TMPDIR/unpack"
+      mkdir "$unpackDir"
+      cd "$unpackDir"
+
+      renamed="$TMPDIR/$name"
       mv "$downloadedFile" "$renamed"
       unpackFile "$renamed"
-    ''
-    # FIXME: handle zip files that contain a single regular file.
-    + lib.optionalString stripRoot ''
+
       shopt -s dotglob
-      if [ "$(ls -d $out/* | wc -l)" != 1 ]; then
-        echo "error: zip file must contain a single directory."
+    ''
+    + (if stripRoot then ''
+      if [ $(find "$unpackDir" -maxdepth 1 -not -path "$unpackDir" | wc -l) != 1 ]; then
```

Comment:
Oh right, I'm using the github split diff view now and I'm not used to it yet.

the previous version is also a bit weird with the `ls -d $out/*`. I think `ls "$unpackDir"` should work fine.

Curious why the switch...